File: leetcode_75/1_Array_String/cpp/334.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution
{
public:
    bool increasingTriplet(vector<int> &nums)
    {
        long int first = 10000000000000000;
        long int second = 10000000000000000;
        for (auto num : nums)
        {
            if (num <= first)
            {
                first = num;
            }
            else if (first < num && num <= second)
            {
                second = num;
            }
            else if (num > second)
            {
                return true;
            }
        }

        return false;
    }
};

File: leetcode_75/Array_String/python/334.py
from typing import List


class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        first, second = float('Inf'), float('Inf')
        for ele in nums:
            if ele <= first:
                first = ele
            elif first < ele <= second:
                second = ele
            elif ele > second:
                return True
        return False


solution = Solution()
solution.increasingTriplet(nums=[1, 2, 3, 4, 5])
